skins/laika/src/components/pages/donation_form/Payment.vue

```diff
@@ -52,7 +53,23 @@ export default Vue.extend( {
 			amount: ( state: any ) => state[ NS_PAYMENT ].values.amount,
 			interval: ( state: any ) => state[ NS_PAYMENT ].values.interval,
 			type: ( state: any ) => state[ NS_PAYMENT ].values.type,
-			disabledPaymentTypes: ( state: any ) => state[ NS_ADDRESS ].addressType === AddressTypeModel.ANON ? [ 'BEZ' ] : [],
+			disabledPaymentTypes: ( state: any ) => {
+				var disabledTypes : String[] = [];
+				if ( state[ NS_ADDRESS ].addressType === AddressTypeModel.ANON ) {
+					disabledTypes.push( 'BEZ' );
+				}
+				if ( state[ NS_PAYMENT ].values.interval !== '0' ) {
+					disabledTypes.push( 'SUB' );
+				}
+				return disabledTypes;
+			},
+			disabledPaymentIntervals: ( state: any ) => {
+				var disabledIntervals : String[] = [];
+				if ( state[ NS_PAYMENT ].values.type === 'SUB' ) {
+					disabledIntervals.push( '1', '3', '6', '12' );
```

Comment:
It's highly unlikely that we'll have different intervals in the future, but in the interest of keeping the amount of places where we list the intervals low, you could do 

    disabledIntervals = this.$props.paymentIntervals.filter( interval => Number(interval) > 0 ).map( interval => String(interval));

or wrap that in a function called `getListOfRecurringIntervals`

Reply:
obviously the better approach, thanks